crates/codegen/spec/src/public/navigation.rs

```diff
@@ -0,0 +1,76 @@
+use std::path::PathBuf;
+
+use anyhow::Result;
+use codegen_utils::context::CodegenContext;
+
+use crate::public::markdown::MarkdownWriter;
+
+pub enum NavigationEntry<'a> {
+    Directory {
+        title: &'a str,
+        path: &'a str,
+        children: Vec<NavigationEntry<'a>>,
+    },
+    Page {
+        title: &'a str,
+        path: &'a str,
+        contents: String,
+    },
+}
+
+impl<'a> NavigationEntry<'a> {
+    pub fn write_to_disk(&self, codegen: &mut CodegenContext, parent_dir: &PathBuf) -> Result<()> {
```

Comment:
Not so keen on this name ... `disk` ? Maybe not a disk ...

Reply:
Changed to `write_files`. How about that?